snuba/subscriptions/worker.py

```diff
@@ -0,0 +1,49 @@
+from __future__ import annotations
+
+from typing import NamedTuple, Mapping, MutableSequence, Optional, Sequence
+
+from snuba.subscriptions.consumer import Tick
+from snuba.subscriptions.data import Subscription
+from snuba.subscriptions.executor import SubscriptionExecutor
+from snuba.subscriptions.scheduler import ScheduledTask, Scheduler
+from snuba.utils.streams.batching import AbstractBatchWorker
+from snuba.utils.streams.producer import Producer
+from snuba.utils.streams.types import Message, Topic
+from snuba.web.query import ClickhouseQueryResult
+
+
+class SubscriptionResult(NamedTuple):
+    task: ScheduledTask[Subscription]
+    result: ClickhouseQueryResult
```

Comment:
The result here only deals with successes right now, by design.

This should unblock #734.